Add missing dependency on KLU when building IDA shared library.

Based on a patch in the Debian sundials package by Dima Kogan.

https://anonscm.debian.org/cgit/debian-science/packages/sundials.git/tree/debian/patches/0001-Added-missing-linkages-

diff --git a/src/ida/CMakeLists.txt b/src/ida/CMakeLists.txt
--- a/src/ida/CMakeLists.txt
+++ b/src/ida/CMakeLists.txt
@@ -135,6 +135,10 @@ IF(BUILD_SHARED_LIBS)
   ADD_LIBRARY(sundials_ida_shared SHARED
     ${ida_SOURCES}  ${ida_BL_SOURCES} ${shared_SOURCES})
 
+  IF(KLU_FOUND)
+    TARGET_LINK_LIBRARIES(sundials_ida_shared -lklu)
+  ENDIF(KLU_FOUND)
+
   # Set the library name and make sure it is not deleted
   SET_TARGET_PROPERTIES(sundials_ida_shared
     PROPERTIES OUTPUT_NAME sundials_ida CLEAN_DIRECT_OUTPUT 1)
